clientconn.go

```diff
@@ -152,6 +152,25 @@ func NewClient(target string, opts ...DialOption) (conn *ClientConn, err error)
 	for _, opt := range opts {
 		opt.apply(&cc.dopts)
 	}
+
+	// Register ClientConn with channelz.
+	cc.channelzRegistration(target)
+	// TODO: Ideally it should be impossible to error from this function after
+	// channelz registration.  This will require removing some channelz logs
+	// from the following functions that can error.  Errors can be returned to
+	// the user, and successful logs can be emitted here, after the checks have
+	// passed and channelz is subsequently registered.
+
+	// Determine the resolver to use.
+	if err := cc.parseTargetAndFindResolver(); err != nil {
+		channelz.RemoveEntry(cc.channelz.ID)
+		return nil, err
+	}
+
+	for _, lateApplyOpt := range globalLateApplyDialOptions {
+		lateApplyOpt.DialOption(cc.parsedTarget.URL).apply(&cc.dopts)
```

Comment:
Nit: `opt` (as above) or even `o` is preferred for very narrowly scoped vars like this.

Reply:
Ah, right. Switched to opt.